platform/image_processor/src/processor/job/resize.rs

```diff
@@ -38,30 +28,17 @@ impl std::fmt::Debug for ImageResizer {
 	}
 }
 
-/// Safety: this function is unsafe because Mat must not outlive self.
-/// The caller must ensure that the returned Mat is dropped before self is
-/// dropped.
-unsafe fn make_mat(buffer: &[rgb::RGBA8], width: usize, height: usize) -> Result<opencv::core::Mat> {
-	assert!(buffer.len() >= width * height, "buffer is too small");
-
-	let data = buffer.as_bytes().as_ptr() as *mut std::ffi::c_void;
-	opencv::core::Mat::new_rows_cols_with_data(
-		height as i32,
-		width as i32,
-		opencv::core::CV_8UC4,
-		data,
-		opencv::core::Mat_AUTO_STEP,
-	)
-	.context("opencv mat new rows cols with data")
-	.map_err(ProcessorError::ImageResize)
-}
-
 impl ImageResizer {
 	pub fn new(target: ImageResizerTarget) -> Self {
 		Self {
+			resizer: fr::Resizer::new(match target.algorithm {
+				ResizeAlgorithm::Nearest => fr::ResizeAlg::Nearest,
+				ResizeAlgorithm::Linear => fr::ResizeAlg::Convolution(fr::FilterType::Bilinear),
+				ResizeAlgorithm::Cubic => fr::ResizeAlg::Convolution(fr::FilterType::CatmullRom),
+				ResizeAlgorithm::Area => fr::ResizeAlg::Convolution(fr::FilterType::Box),
+				ResizeAlgorithm::Lanczos4 => fr::ResizeAlg::Convolution(fr::FilterType::Lanczos3),
```

Comment:
Why is Lanczos4 actually Lanczos3?

Reply:
We can rename the proto, the pure rust impl does not have Lanczos4 only Lanczos3